### PR TITLE
datasource/okta: refactor okta provider

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Refactor the okta directory provider.

* Add the `url` parameter to the `cmd`. It didn't work without this. (for testing outside of the enterprise console)
* Fix the `filterDateFormat`. The sub seconds is required, hence using `.000Z` instead of `.999Z`.
* Switch to the Okta SDK which has built-in support for paging and rate limiting.
* Fix the way groups are synchronized after initial synchronization to only sync changes (for #342)

## Related issues
- #342 

## Checklist
- [x] reference any related issues
- [x] updated unit tests
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: pkg/directory/okta/provider.go [hunk 1]
```diff
@@ -2,266 +2,172 @@ package okta
...
 
-// GetDirectory fetches the groups of which the user is a member
-// https://developer.okta.com/docs/reference/api/users/#get-user-s-groups
+// GetDirectory gets the full directory information for Okta.
 func (p *Provider) GetDirectory(ctx context.Context) ([]directory.Group, []directory.User, error) {
-	groups, err := p.getGroups(ctx)
+	ctx, client, err := okta.NewClient(ctx,
+		okta.WithHttpClientPtr(p.cfg.getHTTPClient()),
+		okta.WithOrgUrl(p.cfg.url),
+		okta.WithTestingDisableHttpsCheck(true),
```
Comment (kralicky) [inline pkg/directory/okta/provider.go:41]: accidentally left in?

File: pkg/directory/okta/provider.go [hunk 2]
```diff
@@ -2,266 +2,172 @@ package okta
...
 	}
 
-	for groupURL != "" {
-		var out []apiGroupObject
-		hdrs, err := p.apiGet(ctx, groupURL, &out)
+	var changedGroups []*okta.Group
+	if lastMembershipUpdated.IsZero() || lastUpdated.IsZero() {
+		// full sync
+		p.groups = make(map[string]okta.Group)
+		p.groupMembers = make(map[string][]okta.User)
```
Comment (kralicky) [inline pkg/directory/okta/provider.go:100]:
```suggestion
		clear(p.groups)
		clear(p.groupMembers)
```

File: pkg/directory/okta/provider.go [hunk 3]
```diff
@@ -2,266 +2,172 @@ package okta
...
+
+		var err error
+		changedGroups, _, err = client.Group.ListGroups(ctx, &query.Params{
+			Limit: int64(p.cfg.batchSize),
+		})
```
Comment (kralicky) [inline pkg/directory/okta/provider.go:105]: does this need logic to handle pagination? something like 
```suggestion
		var err error
		var resp *okta.Response
		changedGroups, resp, err = client.Group.ListGroups(ctx, &query.Params{
			Limit: int64(p.cfg.batchSize),
		})
		if err != nil {
			return fmt.Errorf("error querying all groups: %w", err)
		}

		for resp.HasNextPage() {
			var groups []*okta.Group
			resp, err = resp.Next(ctx, &groups)
			if err != nil {
				return err
			}
			changedGroups = append(changedGroups, groups...)
		}
```

Comment (calebdoxsey) [reply pkg/directory/okta/provider.go:106]: You are correct. I misunderstood the SDK.

File: pkg/directory/okta/provider.go (hunk 1 above)
Comment (calebdoxsey) [reply pkg/directory/okta/provider.go]: ok I switched this to be a config option so we can only add it for tests

**Review by kralicky — APPROVED** [review]

**Pull request merged**